File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/teleport/RandomTeleportSettings.java
package top.likoslupus.cellulosesz.api.teleport;

public record RandomTeleportSettings(
        double centerX,
        double centerZ,
        int minRadius,
        int maxRadius
) {

    public RandomTeleportSettings {
        if (!Double.isFinite(centerX) || !Double.isFinite(centerZ)) {
            throw new IllegalArgumentException("Random teleport center must be finite");
        }
        if (minRadius < 0 || maxRadius <= minRadius) {
            throw new IllegalArgumentException("Random teleport radii must satisfy 0 <= min < max");
        }
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/teleport/RandomTeleportSettings.kt
package top.likoslupus.cellulosesz.api.teleport

@JvmRecord
public data class RandomTeleportSettings(
    public val centerX: Double,
    public val centerZ: Double,
    public val minRadius: Int,
    public val maxRadius: Int
) {

    init {
        require(centerX.isFinite() && centerZ.isFinite()) {
            "Random teleport center must be finite"
        }
        require(minRadius in 0..<maxRadius) {
            "Random teleport radii must satisfy 0 <= min < max"
        }
    }

}
